gateway-service-impl/src/main/java/org/hypertrace/gateway/service/baseline/BaselineServiceQueryParser.java

```diff
@@ -183,13 +171,22 @@ public BaselineEntitiesResponse parseQueryResponse(
               continue;
             }
 
-            Value convertedValue =
-                QueryAndGatewayDtoConverter.convertToGatewayValueForMetricValue(
-                    MetricAggregationFunctionUtil.getValueTypeFromFunction(
-                        timeAggregation.getAggregation(), attributeMetadataMap),
-                    attributeMetadataMap,
-                    metadata,
-                    row.getColumn(i));
+            Value convertedValue;
+            // AVGRATE is adding a specific implementation because Pinot does not directly support this function,
```

Comment:
I'm guessing this is duplicated from elsewhere - can it be shared?

Reply:
done